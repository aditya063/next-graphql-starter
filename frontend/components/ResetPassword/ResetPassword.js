import PropTypes from 'prop-types';
import Router from 'next/router';
import { Mutation } from '@apollo/react-components';

import DisplayError from '../DisplayError/DisplayError';
import DisplaySuccess from '../DisplaySuccess/DisplaySuccess';
import { RESET_PASSWORD_MUTATION } from '../../graphql/queries';
import { passwordRequirements } from '../../constants';
import * as sc from './ResetPassword.style';

class ResetPassword extends React.PureComponent {
  componentDidMount() {
    // Disable submit button at the beginning
    this.props.form.validateFields();
  }

  hasErrors = fieldsError =>
    Object.keys(fieldsError).some(field => fieldsError[field]);

  handleSubmitForm = (e, resetPassword) => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      resetPassword({
        variables: { ...values, resetToken: this.props.resetToken }
      });

      this.props.form.resetFields();
    });
  };

  // handleCompleted = () => Router.push({ pathname: '/' });

  handleError = error => error;

  render() {
    const {
      resetToken,
      resetTokenExpiry,
      form: { getFieldDecorator, getFieldsError, getFieldError, isFieldTouched }
    } = this.props;

    const isTokenPresent = resetToken && resetTokenExpiry;
    const isTokenExpired = Date.now() > resetTokenExpiry;
    const isTokenValid = isTokenPresent && !isTokenExpired;

    const tokenMissingError = {
      message: 'Error: Please submit a new password reset request.'
    };
    const tokenExpiredError = {
      message: 'Your reset request is expired.  Please submit a new one.'
    };

    // Only show error after a field is touched.
    const passwordError = isFieldTouched('password') && getFieldError('password');
    const confirmPasswordError = isFieldTouched('confirmPassword') && getFieldError('confirmPassword');

    return (
      <Mutation
        mutation={RESET_PASSWORD_MUTATION}
        variables={{ ...this.state, resetToken: this.props.resetToken }}
        onError={this.handleError}
        // onCompleted={this.handleCompleted}
      >
        {(resetPassword, { loading, error, data }) => (
          <sc.ResetPassword
            onSubmit={e => this.handleSubmitForm(e, resetPassword)}
          >
            <sc.FormItem
              validateStatus={passwordError ? 'error' : ''}
              help={passwordError || ''}
              hasFeedback
            >
              {getFieldDecorator('password', {
                rules: [
                  { required: true, message: 'Please enter your password' }
                ]
              })(
                <sc.InputPassword
                  placeholder="password"
                  onPressEnter={e => this.handleSubmitForm(e, resetPassword)}
                  prefix={(
                    <sc.InputIcon
                      type="lock"
                      style={{ color: 'rgba(0,0,0,.25)' }}
                    />
)}
                />
              )}
            </sc.FormItem>

            <sc.FormItem
              validateStatus={confirmPasswordError ? 'error' : ''}
              help={confirmPasswordError || ''}
              hasFeedback
            >
              {getFieldDecorator('confirmPassword', {
                rules: [
                  { required: true, message: 'Please confirm your password' }
                ]
              })(
                <sc.InputConfirmPassword
                  placeholder="confirm password"
                  onPressEnter={e => this.handleSubmitForm(e, resetPassword)}
                  prefix={(
                    <sc.InputIcon
                      type="lock"
                      style={{ color: 'rgba(0,0,0,.25)' }}
                    />
)}
                />
              )}
            </sc.FormItem>

            {!isTokenPresent && <DisplayError error={tokenMissingError} />}

            {isTokenPresent && isTokenExpired && (
              <DisplayError error={tokenExpiredError} />
            )}

            {isTokenValid && error && <DisplayError error={error} />}

            {data && data.resetPassword && (
              <DisplaySuccess message="Your password has been successfully changed." />
            )}

            <sc.TypographyText strong>
              {passwordRequirements.title}
            </sc.TypographyText>

            <sc.PassList
              split={false}
              dataSource={passwordRequirements.reqs}
              renderItem={item => (
                <sc.PassListItem>
                  <sc.ListIcon type="minus" />
                  <sc.TypographyText>{item}</sc.TypographyText>
                </sc.PassListItem>
              )}
            />

            <sc.FormItem>
              <sc.SubmitBtn
                loading={loading}
                type="primary"
                htmlType="submit"
                disabled={this.hasErrors(getFieldsError())}
              >
                Submit
              </sc.SubmitBtn>
            </sc.FormItem>
          </sc.ResetPassword>
        )}
      </Mutation>
    );
  }
}

ResetPassword.defaultProps = {
  resetToken: '',
  resetTokenExpiry: ''
};

ResetPassword.propTypes = {
  resetToken: PropTypes.string,
  resetTokenExpiry: PropTypes.string,
  form: PropTypes.shape({
    validateFields: PropTypes.func.isRequired,
    getFieldDecorator: PropTypes.func.isRequired,
    getFieldsError: PropTypes.func.isRequired,
    getFieldError: PropTypes.func.isRequired,
    isFieldTouched: PropTypes.func.isRequired,
    resetFields: PropTypes.func.isRequired
  }).isRequired
};

export default sc.ResetPassword.create({ name: 'ResetPassword' })(
  ResetPassword
);
