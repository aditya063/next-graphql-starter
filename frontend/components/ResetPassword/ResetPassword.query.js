import gql from 'graphql-tag';

export const RESET_PASSWORD_MUTATION = gql`
  mutation RESET_PASSWORD_MUTATION(
    $resetToken: String!
    $password: String!
    $confirmPassword: String!
  ) {
    resetPassword(
      resetToken: $resetToken
      password: $password
      confirmPassword: $confirmPassword
    ) {
      id
      email
    }
  }
`;

export const MOCK_RESET_PASSWORD_MUTATION = {
  request: {
    query: RESET_PASSWORD_MUTATION,
    variables: {
      resetToken: '4e4a8fb6e44ec32642cfa410243652f85885bc72',
      password: 'mockpass',
      confirmPassword: 'mockpass'
    }
  },
  result: {
    data: {
      resetPassword: [
        {
          id: '1',
          email: 'mock@email.com'
        }
      ]
    }
  }
};
