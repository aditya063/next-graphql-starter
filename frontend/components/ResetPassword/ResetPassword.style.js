import { form } from '../../styles/form.style';

export { form };
