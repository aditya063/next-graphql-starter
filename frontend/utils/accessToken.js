import fetch from 'isomorphic-unfetch';
// import cookie from 'cookie';
// import jwt from 'jsonwebtoken';

let accessToken = '';

export const setAccessToken = token => (accessToken = token);

export const getAccessToken = () => accessToken;

export const clearAccessToken = () => (accessToken = '');

export const fetchAccessToken = async refreshToken => {
  const url =
    process.env.NODE_ENV === 'development'
      ? process.env.DEV_REFRESH_URL
      : process.env.PROD_REFRESH_URL;

  const options = {
    method: 'GET',
    credentials: 'include',
    headers: { cookie: `rt=${refreshToken}` }
  };

  try {
    const response = await fetch(url, options);

    const data = await response.json();

    return data.accessToken;
  } catch (error) {
    if (process.env.NODE_ENV === 'development')
      console.error('fetchAccessToken error: ', error);
  }
};

// export const verifyAccessToken = accessToken => {
//   try {
//     jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET);
//   } catch (error) {
//     if (process.env.NODE_ENV === 'development')
//       console.error('fetchAccessToken error: ', error);
//   }
// };

// const checkAccessToken = async cookies => {
//   const accessToken = getAccessToken();

//   if (!accessToken) return true;

//   const parsedCookies = cookie.parse(cookies);

//   if (!parsedCookies.rt) return '';

//   // const response = await fetchAccessToken(parsedCookies.rt);

//   // const data = await response.json();

//   // setAccessToken(data.accessToken);
//   // setAccessTokenExpiry(data.exp);

//   // return data.accessToken;
// };
