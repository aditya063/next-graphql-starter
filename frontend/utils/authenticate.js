import jwt from 'jsonwebtoken';

const authenticate = cookie => {
  const secret = process.env.JWT_SECRET;
  const token = cookie.slice(6);

  try {
    jwt.verify(token, secret);

    return true;
  } catch {
    return false;
  }
};

const setCache = (cache, isAuthenticated) => {
  cache.writeData({
    data: {
      isLoggedIn: isAuthenticated
    }
  });
};

export default (cache, cookie) => {
  const isAuthenticated = cookie ? authenticate(cookie) : false;

  setCache(cache, isAuthenticated);
};
