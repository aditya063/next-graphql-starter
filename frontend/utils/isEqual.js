// compare props

export default (prevProps, nextProps) => {
  const obj1 = prevProps;
  console.log('TCL: isEqual -> obj1', obj1);
  const obj2 = nextProps;
  console.log('TCL: isEqual -> obj2', obj2);

  const diff = Object.keys(obj1).reduce((result, key) => {
    if (!Object.prototype.hasOwnProperty.call(obj2, key)) {
      result.push(key);
    } else if (_.isEqual(obj1[key], obj2[key])) {
      const resultKeyIndex = result.indexOf(key);

      result.splice(resultKeyIndex, 1);
    }

    return result;
  }, Object.keys(obj2));

  console.log('TCL: areEqual -> diff', diff);

  const areEqual = !diff;

  console.log('TCL: areEqual -> areEqual', areEqual);

  return areEqual;
};
