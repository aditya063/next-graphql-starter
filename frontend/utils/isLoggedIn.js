/* eslint-disable no-console */
import * as query from '../components/wrappers/WithUser/WithUser.query';

export const isLoggedIn = async client => {
  try {
    const res = await client.query({
      query: query.ME_QUERY
    });

    return res.data.me;
  } catch (err) {
    return null;
  }
};
