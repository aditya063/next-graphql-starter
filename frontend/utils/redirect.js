import Router from 'next/router';

export const redirect = (context, target) => {
  if (context.res) {
    context.res.writeHead(303, { Location: target });
    context.res.end();
  } else {
    Router.replace(target);
  }
};
