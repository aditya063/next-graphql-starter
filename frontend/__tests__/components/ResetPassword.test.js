import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { ThemeProvider } from 'styled-components';
import { load } from '../../utils/load';
import ResetPassword from '../../components/ResetPassword/ResetPassword';
// eslint-disable-next-line max-len
import { MOCK_RESET_PASSWORD_MUTATION } from '../../components/ResetPassword/ResetPassword.query';
import theme from '../../styles/theme.style';

describe('ResetPassword', () => {
  let mockProps;
  let mockQueries;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    mockQueries = [MOCK_RESET_PASSWORD_MUTATION];
    wrapper = mount(
      <MockedProvider mocks={mockQueries} addTypename={false}>
        <ThemeProvider theme={theme}>
          <ResetPassword {...mockProps} />
        </ThemeProvider>
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('matches snapshot', () => {
    const wrapSnap = wrapper.find('formstyle__form');

    // console.log(wrapper.debug());

    expect(wrapSnap).toMatchSnapshot();
  });

  it('matches snapshot - loaded', async () => {
    const wrapSnap = wrapper.find('formstyle__form');

    await load(wrapper);

    // console.log(wrapper.debug());

    expect(wrapSnap).toMatchSnapshot();
  });
});
