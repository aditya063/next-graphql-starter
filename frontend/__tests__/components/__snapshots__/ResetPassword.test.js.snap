// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword matches snapshot - loaded 1`] = `
.c0 > fieldset {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border: 1px solid #393939;
  padding: 4px;
  border-radius: 2px;
  margin: 5px;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: 1px 1px 0 #fff;
  font-size: 1rem;
  font-weight: 600;
  padding: 0.3rem 1rem;
  border: 2px solid;
  border-radius: 3px;
  cursor: pointer;
  -webkit-transition: background-color 0.3s ease-out;
  transition: background-color 0.3s ease-out;
  background: none;
  white-space: nowrap;
  margin-top: 10px;
  margin-left: auto;
}

.c4:hover {
  background-color: #eee;
}

.c4:active {
  box-shadow: 0 2px 6px rgba(0,0,0,0.25) inset;
}

.c2 {
  margin-bottom: 0;
  font-size: 1rem;
}

.c3 {
  margin: 5px 0 0 0;
  padding-left: 20px;
}

<formstyle__form
  onSubmit={[Function]}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "formstyle__form-q97ky5-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "> fieldset{display:flex;}",
          ],
        },
        "displayName": "formstyle__form",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "formstyle__form-q97ky5-0",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    onSubmit={[Function]}
  >
    <form
      className="c0"
      onSubmit={[Function]}
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Reset Your Password
        </h2>
        <label
          htmlFor="password"
        >
          Password
          <ResetPasswordstyle__inputText
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "border:1px solid ",
                      [Function],
                      ";padding:4px;border-radius:2px;",
                      "margin:5px;",
                    ],
                  },
                  "displayName": "ResetPasswordstyle__inputText",
                  "foldedComponentIds": Array [
                    "inputstyle__input-bgiqmz-0",
                  ],
                  "render": [Function],
                  "styledComponentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            >
              <input
                className="c1"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />
            </StyledComponent>
          </ResetPasswordstyle__inputText>
        </label>
        <label
          htmlFor="confirmPassword"
        >
          Confirm Password
          <ResetPasswordstyle__inputText
            name="confirmPassword"
            onChange={[Function]}
            placeholder="confirmPassword"
            type="password"
            value=""
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "border:1px solid ",
                      [Function],
                      ";padding:4px;border-radius:2px;",
                      "margin:5px;",
                    ],
                  },
                  "displayName": "ResetPasswordstyle__inputText",
                  "foldedComponentIds": Array [
                    "inputstyle__input-bgiqmz-0",
                  ],
                  "render": [Function],
                  "styledComponentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              name="confirmPassword"
              onChange={[Function]}
              placeholder="confirmPassword"
              type="password"
              value=""
            >
              <input
                className="c1"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="confirmPassword"
                type="password"
                value=""
              />
            </StyledComponent>
          </ResetPasswordstyle__inputText>
        </label>
        <ResetPasswordstyle__h3PassTitle>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ResetPasswordstyle__h3PassTitle-u9xmn6-2",
                  "isStatic": true,
                  "lastClassName": "c2",
                  "rules": Array [
                    "margin-bottom:0;font-size:1rem;",
                  ],
                },
                "displayName": "ResetPasswordstyle__h3PassTitle",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ResetPasswordstyle__h3PassTitle-u9xmn6-2",
                "target": "h3",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h3
              className="c2"
            >
              Password must contain:
            </h3>
          </StyledComponent>
        </ResetPasswordstyle__h3PassTitle>
        <ResetPasswordstyle__ulPassList
          aria-label="Password must contain:"
        >
          <StyledComponent
            aria-label="Password must contain:"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ResetPasswordstyle__ulPassList-u9xmn6-3",
                  "isStatic": true,
                  "lastClassName": "c3",
                  "rules": Array [
                    "margin:5px 0 0 0;padding-left:20px;",
                  ],
                },
                "displayName": "ResetPasswordstyle__ulPassList",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ResetPasswordstyle__ulPassList-u9xmn6-3",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <ul
              aria-label="Password must contain:"
              className="c3"
            >
              <li>
                at least 8 charactors
              </li>
              <li>
                an uppercase letter
              </li>
              <li>
                a lowercase letter
              </li>
              <li>
                a number
              </li>
            </ul>
          </StyledComponent>
        </ResetPasswordstyle__ulPassList>
        <ResetPasswordstyle__inputBtn
          disabled={true}
          type="submit"
          value="Reset Your Password"
        >
          <StyledComponent
            disabled={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ResetPasswordstyle__inputBtn-u9xmn6-1",
                  "isStatic": false,
                  "lastClassName": "c4",
                  "rules": Array [
                    "text-decoration:none;text-shadow:1px 1px 0 #fff;font-size:1rem;font-weight:600;padding:0.3rem 1rem;border:2px solid;border-radius:3px;cursor:pointer;transition:background-color 0.3s ease-out;background:none;white-space:nowrap;&:hover{background-color:#eee;}&:active{box-shadow:0 2px 6px rgba(0,0,0,0.25) inset;}",
                    "color:",
                    [Function],
                    ";",
                    "margin-top:10px;margin-left:auto;",
                  ],
                },
                "displayName": "ResetPasswordstyle__inputBtn",
                "foldedComponentIds": Array [
                  "buttonstyle__inputBtn-sc-5slzqu-3",
                  "buttonstyle__submitInputBtn-sc-5slzqu-5",
                ],
                "render": [Function],
                "styledComponentId": "ResetPasswordstyle__inputBtn-u9xmn6-1",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            type="submit"
            value="Reset Your Password"
          >
            <input
              className="c4"
              disabled={true}
              type="submit"
              value="Reset Your Password"
            />
          </StyledComponent>
        </ResetPasswordstyle__inputBtn>
      </fieldset>
    </form>
  </StyledComponent>
</formstyle__form>
`;

exports[`ResetPassword matches snapshot 1`] = `
.c0 > fieldset {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border: 1px solid #393939;
  padding: 4px;
  border-radius: 2px;
  margin: 5px;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: 1px 1px 0 #fff;
  font-size: 1rem;
  font-weight: 600;
  padding: 0.3rem 1rem;
  border: 2px solid;
  border-radius: 3px;
  cursor: pointer;
  -webkit-transition: background-color 0.3s ease-out;
  transition: background-color 0.3s ease-out;
  background: none;
  white-space: nowrap;
  margin-top: 10px;
  margin-left: auto;
}

.c4:hover {
  background-color: #eee;
}

.c4:active {
  box-shadow: 0 2px 6px rgba(0,0,0,0.25) inset;
}

.c2 {
  margin-bottom: 0;
  font-size: 1rem;
}

.c3 {
  margin: 5px 0 0 0;
  padding-left: 20px;
}

<formstyle__form
  onSubmit={[Function]}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "formstyle__form-q97ky5-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "> fieldset{display:flex;}",
          ],
        },
        "displayName": "formstyle__form",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "formstyle__form-q97ky5-0",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    onSubmit={[Function]}
  >
    <form
      className="c0"
      onSubmit={[Function]}
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Reset Your Password
        </h2>
        <label
          htmlFor="password"
        >
          Password
          <ResetPasswordstyle__inputText
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "border:1px solid ",
                      [Function],
                      ";padding:4px;border-radius:2px;",
                      "margin:5px;",
                    ],
                  },
                  "displayName": "ResetPasswordstyle__inputText",
                  "foldedComponentIds": Array [
                    "inputstyle__input-bgiqmz-0",
                  ],
                  "render": [Function],
                  "styledComponentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            >
              <input
                className="c1"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />
            </StyledComponent>
          </ResetPasswordstyle__inputText>
        </label>
        <label
          htmlFor="confirmPassword"
        >
          Confirm Password
          <ResetPasswordstyle__inputText
            name="confirmPassword"
            onChange={[Function]}
            placeholder="confirmPassword"
            type="password"
            value=""
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "border:1px solid ",
                      [Function],
                      ";padding:4px;border-radius:2px;",
                      "margin:5px;",
                    ],
                  },
                  "displayName": "ResetPasswordstyle__inputText",
                  "foldedComponentIds": Array [
                    "inputstyle__input-bgiqmz-0",
                  ],
                  "render": [Function],
                  "styledComponentId": "ResetPasswordstyle__inputText-u9xmn6-0",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              name="confirmPassword"
              onChange={[Function]}
              placeholder="confirmPassword"
              type="password"
              value=""
            >
              <input
                className="c1"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="confirmPassword"
                type="password"
                value=""
              />
            </StyledComponent>
          </ResetPasswordstyle__inputText>
        </label>
        <ResetPasswordstyle__h3PassTitle>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ResetPasswordstyle__h3PassTitle-u9xmn6-2",
                  "isStatic": true,
                  "lastClassName": "c2",
                  "rules": Array [
                    "margin-bottom:0;font-size:1rem;",
                  ],
                },
                "displayName": "ResetPasswordstyle__h3PassTitle",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ResetPasswordstyle__h3PassTitle-u9xmn6-2",
                "target": "h3",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h3
              className="c2"
            >
              Password must contain:
            </h3>
          </StyledComponent>
        </ResetPasswordstyle__h3PassTitle>
        <ResetPasswordstyle__ulPassList
          aria-label="Password must contain:"
        >
          <StyledComponent
            aria-label="Password must contain:"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ResetPasswordstyle__ulPassList-u9xmn6-3",
                  "isStatic": true,
                  "lastClassName": "c3",
                  "rules": Array [
                    "margin:5px 0 0 0;padding-left:20px;",
                  ],
                },
                "displayName": "ResetPasswordstyle__ulPassList",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "ResetPasswordstyle__ulPassList-u9xmn6-3",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <ul
              aria-label="Password must contain:"
              className="c3"
            >
              <li>
                at least 8 charactors
              </li>
              <li>
                an uppercase letter
              </li>
              <li>
                a lowercase letter
              </li>
              <li>
                a number
              </li>
            </ul>
          </StyledComponent>
        </ResetPasswordstyle__ulPassList>
        <ResetPasswordstyle__inputBtn
          disabled={true}
          type="submit"
          value="Reset Your Password"
        >
          <StyledComponent
            disabled={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "ResetPasswordstyle__inputBtn-u9xmn6-1",
                  "isStatic": false,
                  "lastClassName": "c4",
                  "rules": Array [
                    "text-decoration:none;text-shadow:1px 1px 0 #fff;font-size:1rem;font-weight:600;padding:0.3rem 1rem;border:2px solid;border-radius:3px;cursor:pointer;transition:background-color 0.3s ease-out;background:none;white-space:nowrap;&:hover{background-color:#eee;}&:active{box-shadow:0 2px 6px rgba(0,0,0,0.25) inset;}",
                    "color:",
                    [Function],
                    ";",
                    "margin-top:10px;margin-left:auto;",
                  ],
                },
                "displayName": "ResetPasswordstyle__inputBtn",
                "foldedComponentIds": Array [
                  "buttonstyle__inputBtn-sc-5slzqu-3",
                  "buttonstyle__submitInputBtn-sc-5slzqu-5",
                ],
                "render": [Function],
                "styledComponentId": "ResetPasswordstyle__inputBtn-u9xmn6-1",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            type="submit"
            value="Reset Your Password"
          >
            <input
              className="c4"
              disabled={true}
              type="submit"
              value="Reset Your Password"
            />
          </StyledComponent>
        </ResetPasswordstyle__inputBtn>
      </fieldset>
    </form>
  </StyledComponent>
</formstyle__form>
`;
